sdk/go/pulumi-language-go/main.go

```diff
@@ -77,7 +82,17 @@ func findProgram(binary string) (*exec.Cmd, error) {
 		return nil, errors.Errorf("Failed to find go files for 'go run' matching %s", goFileSearchPattern)
 	}
 
-	return exec.Command(program, "run", cwd), nil
+	binaryToRun := path.Join(cwd, PULUMI_BINARY_NAME)
+	buildCmd := exec.Command(program, "build", "-o", binaryToRun, cwd)
+	if err := buildCmd.Run(); err != nil {
+		logging.V(5).Infof("Unable to `go build` falling back to `go run`")
+		// fallback to old behavior if failure. i.e.
+		// - compilation failure
+		// - directory is not writable
```

Comment:
it was a pattern at my previous company that directories and files were not writable unless necessary. we can fallback on `go run` to avoid this.

Reply:
temp should always be writable, go run already writes it's compiled object to temp, we should do the same.